system/CodeIgniter.php

```diff
@@ -309,7 +316,11 @@ public function run(?RouteCollectionInterface $routes = null, bool $returnRespon
         if ($this->request instanceof IncomingRequest && strtolower($this->request->getMethod()) === 'cli') {
             $this->response->setStatusCode(405)->setBody('Method Not Allowed');
 
-            $this->sendResponse();
+            if (! $this->returnResponse) {
+                $this->sendResponse();
+            } else {
+                return $this->response;
+            }
 
             return;
```

Comment:
this can be changed to:

```suggestion
            if ($this->returnResponse) {
                return $this->response;
            }

            $this->sendResponse();
            return;
```

Reply:
Yes, that makes it a bit easier to read!